hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -217,11 +219,11 @@ public R put(T key, R value) {
       this.currentInMemoryMapSize += this.estimatedPayloadSize;
       // Remove the old version of the record from disk first to avoid data duplication.
       if (inDiskContainsKey(key)) {
-        getDiskBasedMap().remove(key);
+        getDiskBasedMap(true).remove(key);
```

Comment:
I see we are setting the arg to true only here. 
do you think we can introduce a overloaded method for getDiskBasedMap() which sets the forceInitialization to false. so that we don't need to fix umpteen callers.

Reply:
So `getDiskBasedMap(false)` becomes just `getDiskBasedMap()` and then something like `getOrCreateDiskBasedMap()` for the `true` case?

Reply:
+1 to `getOrCreateDiskBasedMap()`